AnkiHubSoftware/ankihub_addon
# [BUILD-105] feat: Add "Copy Anki Note ID to clipboard" context menu action

On the suggestion dialog for suggesting note deletions, there will be a field for providing the Anki note id of a duplicate note. 
However there is a problem: Currently it's not easy to get the Anki note id of a note from the Anki browser (without the help of other add-ons).
In this task we will add a browser context menu action for copying the Anki note id of a note to the clipboard.

## Related issues
https://ankihub.atlassian.net/browse/BUILD-105
https://ankihub.atlassian.net/browse/BUILD-89?focusedCommentId=10162

## Proposed changes
- Refactor the `_on_browser_will_show_context_menu` function to make it easier to add the new context menu action
- Add the `AnkiHub: Copy Anki Note ID to clipboard` context menu action
- Add tests. The `_on_browser_will_show_context_menu` function wasn't covered by tests yet and this PR adds some tests.

## How to reproduce
- Open the Anki Browser
- Select an AnkiHub note and right click it
- Click the `AnkiHub: Copy Anki Note ID to clipboard` item in the menu
- Verify that the Anki note id is now in your clipboard. For another way to get the Anki note id, you can click the note -> hover over the `Cards` item in the menu (only if the browser is in "notes-mode") -> Click `Card Info...` -> the dialog that opens shows the anki note id as well

## Screenshots
<img src="https://github.com/ankipalace/ankihub_addon/assets/31575114/74f09607-2f68-485c-9a3a-de743c61ae29" width="800" />
